src/DynamoCore/Library/MemberDocumentNode.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace Dynamo.DSEngine
+{
+    class MemberDocumentNode
+    {
+        private readonly string fullyQualifiedName;
+        private string summary;
+        private string searchTags;
+        private readonly Dictionary<string, string> parameters;
+
+        public string FullyQualifiedName { get { return fullyQualifiedName; } }
+        public string Summary 
+        {
+            get
+            {
+                if (String.IsNullOrEmpty(summary))
+                    return String.Empty;
+                return summary;
+            }
+            set
+            {
+                summary = value;
+            }
+        }
+        public string SearchTags
+        {
+            get
+            {
+                if (String.IsNullOrEmpty(searchTags))
+                    return String.Empty;
+                return searchTags;
+            }
+            set
+            {
+                searchTags = value;
+            }
+        }
```

Comment:
Let us not accept a `null` value here at all, so `summary` will never be `null`. Same thing for `searchTags` (they can both be empty strings):

```csharp
public string Summary 
{
    get
    {
        return summary;
    }
    set
    {
        if (value == null)
            throw new ArgumentNullException("value");
        summary = value;
    }
}
```